server/controllers/v1/post.ts

```diff
@@ -105,6 +106,81 @@ export default class PostController {
     };
   }
 
+  async searchForPosts(
+    userID: string,
+    query: string,
+    limit: number,
+    offset: number
+  ): Promise<{
+    status: number;
+    data: { result?: PostUserPreview[]; count: number; total: number };
+  }> {
+    // await db.sequelize.query(`DROP TABLE IF EXISTS "PostSearches";`);
+    // await db.sequelize.query(
+    //   `CREATE TEMP TABLE "PostSearches" AS
+    //   SELECT *, (
+    //     setweight(to_tsvector(coalesce("title",'')), 'A') ||
+    //     setweight(to_tsvector(coalesce("location",'')), 'C') ||
+    //     setweight(to_tsvector(coalesce("body",'')), 'D')
+    //   ) as "tsvector"
+    //   FROM "Posts";`
+    // );
+    const data = await Promise.all([
+      db.sequelize.query(
+        `SELECT 
+          "Post"."id", 
+          "Post"."body", 
+          "Post"."title", 
+          "Post"."createdAt", 
+          "Post"."thumbnail", 
+          (SELECT COUNT(*) FROM "UserPostLikes" AS "Likes" 
+            WHERE "Likes"."postID" = "Post"."id") AS "likeCount", 
+          (SELECT COUNT(*) FROM "UserPostLikes" AS "Likes" 
+            WHERE "Likes"."postID" = "Post"."id" AND "Likes"."userID" = $1) AS "doesUserLike", 
+          json_build_object(
+            'firstName', "User"."firstName", 
+            'lastName', "User"."lastName",
+            'id', "User"."id"
+          ) AS "User",
+          (
+            setweight(to_tsvector(coalesce("Post"."title",'')), 'A') || 
```

Comment:
Can we also search by an author? I.e username or firstname lastname? If you implement that, can you update `My Posts` (on the menu bar) to search by the current user